File: pos-backend/src/main/java/com/developer/pos/v2/catalog/application/dto/QrMenuDto.java
package com.developer.pos.v2.catalog.application.dto;

import java.util.List;

public record QrMenuDto(
        Long storeId,
        String storeCode,
        String storeName,
        List<CategoryDto> categories
) {
    public record CategoryDto(
            Long categoryId,
            String categoryCode,
            String categoryName,
            List<MenuItemDto> items
    ) {
    }

    public record MenuItemDto(
            Long productId,
            String productCode,
            String productName,
            Long skuId,
            String skuCode,
            String skuName,
            long unitPriceCents
    ) {
    }
}

File: android-pos/app/src/main/java/com/developer/pos/data/remote/dto/QrMenuDto.kt
package com.developer.pos.data.remote.dto

data class QrMenuDto(
    val storeId: Long,
    val storeCode: String,
    val storeName: String,
    val categories: List<QrMenuCategoryDto>
)

data class QrMenuCategoryDto(
    val categoryId: Long,
    val categoryCode: String,
    val categoryName: String,
    val items: List<QrMenuItemDto>
)

data class QrMenuItemDto(
    val skuId: Long,
    val skuCode: String,
    val skuName: String,
    val unitPriceCents: Long,
    val categoryId: Long
)
